File: infrastructure/src/main/java/com/study/platform/global/oauth/kakao/dto/KakaoTokenResponse.java
package com.study.platform.global.oauth.kakao.dto;


import com.fasterxml.jackson.annotation.JsonProperty;

public record KakaoTokenResponse(

        @JsonProperty("access_token")
        String accessToken,

        @JsonProperty("token_type")
        String tokenType,

        @JsonProperty("refresh_token")
        String refreshToken,

        @JsonProperty("expires_in")
        int expiresIn,

        @JsonProperty("id_token")
        String idToken
) {}

File: infrastructure/src/main/kotlin/com/study/platform/global/oauth/kakao/dto/KakaoTokenResponse.kt
package com.study.platform.global.oauth.kakao.dto

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class KakaoTokenResponse @JsonCreator constructor(
    @JsonProperty("access_token")
    val accessToken: String,

    @JsonProperty("token_type")
    val tokenType: String,

    @JsonProperty("refresh_token")
    val refreshToken: String,

    @JsonProperty("expires_in")
    val expiresIn: Int,

    @JsonProperty("id_token")
    val idToken: String
)
